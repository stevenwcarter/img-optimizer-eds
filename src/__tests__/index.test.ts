import "@testing-library/jest-dom";
import createMultiOptimizedPicture, { PictureProps } from "..";

describe("createMultiOptimizedPicture", () => {
  let pic: PictureProps;
  let pic1: PictureProps;
  let pic2: PictureProps;
  beforeEach(() => {
    pic = {
      src: "https://test.com/img.jpg",
    };
    pic1 = {
      src: "https://test.com/img2.jpg",
      breakpoints: [{ width: 1000 }],
    };
    pic2 = {
      src: "https://test.com/img.jpg",
      breakpoints: [{ media: "(min-width: 600px)", width: 2000 }],
    };
  });
  afterEach(() => {
    jest.restoreAllMocks();
  });
  it("matches snapshot w eager=lazy", () => {
    const result = createMultiOptimizedPicture([pic]);

    expect(result).toMatchSnapshot();
  });
  it("matches snapshot w eager=true", () => {
    pic.eager = true;
    const result = createMultiOptimizedPicture([pic]);

    expect(result).toMatchSnapshot();
  });
  it("mimics the existing version with only a single element", () => {
    const result = createMultiOptimizedPicture([pic]);

    expect(result!.querySelectorAll("source").length).toBe(3);
  });
  it("multiple photos has correct number of `source` elements", () => {
    const result = createMultiOptimizedPicture([pic1, pic2]);

    expect(result!.querySelectorAll("source").length).toBe(2);
  });
  it("multiple photos has only one `img` tag", () => {
    const result = createMultiOptimizedPicture([pic1, pic2]);

    expect(result!.querySelectorAll("img").length).toBe(1);
  });
  it("multiple photos matches snapshot", () => {
    const result = createMultiOptimizedPicture([pic1, pic2]);

    expect(result).toMatchSnapshot();
  });
  it("should return null with empty array as input", () => {
    const consoleSpy = jest
      .spyOn(console, "error")
      .mockImplementation(() => {});
    expect(createMultiOptimizedPicture([])).toBeNull();
    expect(consoleSpy).toHaveBeenCalledWith("No pictures provided");
  });
});
