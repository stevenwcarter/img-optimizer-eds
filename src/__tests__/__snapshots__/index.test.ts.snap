// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMultiOptimizedPicture matches snapshot w eager=lazy 1`] = `
<picture>
  <source
    media="(min-width: 600px)"
    srcset="/img.jpg?width=2000&format=webply&optimize=medium"
    type="image/webp"
  />
  <source
    media="(min-width: 600px)"
    srcset="/img.jpg?width=2000&format=jpg&optimize=medium"
  />
  <source
    srcset="/img.jpg?width=750&format=webply&optimize=medium"
    type="image/webp"
  />
  <img
    alt="Alt text"
    loading="lazy"
    src="/img.jpg?width=750&format=jpg&optimize=medium"
  />
</picture>
`;

exports[`createMultiOptimizedPicture matches snapshot w eager=true 1`] = `
<picture>
  <source
    media="(min-width: 600px)"
    srcset="/img.jpg?width=2000&format=webply&optimize=medium"
    type="image/webp"
  />
  <source
    media="(min-width: 600px)"
    srcset="/img.jpg?width=2000&format=jpg&optimize=medium"
  />
  <source
    srcset="/img.jpg?width=750&format=webply&optimize=medium"
    type="image/webp"
  />
  <img
    alt="Alt text"
    loading="eager"
    src="/img.jpg?width=750&format=jpg&optimize=medium"
  />
</picture>
`;

exports[`createMultiOptimizedPicture multiple photos matches snapshot 1`] = `
<picture>
  <source
    media="(min-width: 600px)"
    srcset="/img.jpg?width=2000&format=webply&optimize=medium"
    type="image/webp"
  />
  <source
    srcset="/img2.jpg?width=1000&format=webply&optimize=medium"
    type="image/webp"
  />
  <img
    alt="Alt text"
    loading="lazy"
    src="/img.jpg?width=2000&format=jpg&optimize=medium"
  />
</picture>
`;
